### DISCUSSION TITLE
Favorites save without closing screen

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#516] [Q&A] [answered]
I'm sure there will be pro and cons but....

When I'm in the menu. i.e. "my Collection", "Favorites" and for example I change the symbol of one favorite and press the save symbol,
the menu completely closes after that. 

If I will change another favorite I have to click "through" and "in" again....a little bit like a click orgy :-(

My recommendation is: Either after saving, the menu remains open like it was before saving and in the same place, or maximum a jump one level above should happen then.
A Menu should only close on demand or by going level wise "out" again.

Cheers, Dirk

**Comment by r11d11** [comment]
I have also often been annoyed that the Save button closes everything. I would have expected the Save button to save and that's it. If I want to close the menu, I press the x or swipe from right to left, but not when saving.

**Comment by devemux86** [comment]
@DIrgi thanks for the suggestion

The save in favorites screen is like the OK button in other screens, e.g. profiles, waypoints, etc.
Pressing it, saves and activates all user changes. But we could think for an alternative way to work.

**Reply by bios4** [reply]
> Pressing it, saves and activates all user changes.

Like it should - can you achieve this in the same way but just not "close down' the "My collections" screen but keep everything visible "as is"?

**Reply by devemux86** [reply]
It is possible the save button in the favorites screen to just save the user changes and not close the screen.
Do we want to change it to work this way?

Note: it is not necessary to press save after each change. Can make all changes and press save at the end.

**Reply by DIrgi** [reply]
you know this as programmer. But a user first don’t know…

I would like to do not close the menu by saving. And I would like to have a short popup feedback one the save button that it is successfully saved.
This would be nice.

**Reply by devemux86** [reply]
Both will be available in the next update.

**Reply by DIrgi** [reply]
I know…always demanding but…is it possible to have as well a undo function to restore the previous condition?

**Reply by devemux86** [reply]
> I know…always demanding but…

No problem, this is the way to make Cruiser better and more user-friendly!

> is it possible to have as well a undo function to restore the previous condition?

The changes could be too many and different to keep history for all of them.
This could be possible, but it is not an easy task. :slightly_smiling_face:

**Reply by DIrgi** [reply]
Hmm… I see… question is how do you store and grant settings on next reload? Don‘t you have a kind of a settings file? And if so, pressing „save“ does a backup of this and undo reload it again 🤔?

**Reply by devemux86** [reply]
Favorites are saved in multiple files, so it is not so simple.
And it is not just the storage, but also the user interface, map, etc.

**Comment by devemux86** [comment]
@DIrgi please do not close the topics of "Ideas" as they are hidden in the discussion list.
We can usually close topics of releases or beta versions or questions / reports.

**Reply by DIrgi** [reply]
I closed it because for me it was „solved“ since there was no „answered“ existing like in Q&A but it was an idea and not Q&A.
Finally I don’t mind but don’t blame me then when I don’t mark anything as „solved“.
(I know it from other forums where people being asked to close the case once it’s solved.)

**Reply by devemux86** [reply]
No problem! :slightly_smiling_face: 

Ideas can remain open until they are implemented in the public release on Google Play.
Until then we can still discuss and change their implementation in the Beta versions.

**Comment by devemux86** [comment, marked as answer]
> I'm sure there will be pro and cons but....

Indeed it can cause problems, so it's not recommended.
